helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml

```diff
@@ -50,6 +50,62 @@ spec:
         service: kubelet
       record: slo_target
 
+    # -- managed-apps
+    # record when pods of a daemonset with label "label_giantswarm_io_monitoring_basic_sli" are down
+    - expr: |
+        label_replace(kube_daemonset_status_number_unavailable
+        and on(daemonset,cluster_id,cluster_type,namespace, workload_name)
+        kube_daemonset_labels{label_giantswarm_io_monitoring_basic_sli='true'}, "service", "$1", "workload_name", "(.*)" )
```

Comment:
Why do we copy `workload_name` to `service`? Is it a convention? I have a doubt: what if a single workload exposes multiple k8s API `Service` objects? Is not misleading?

Reply:
We use the `service` label to identify each service in the slo_ metrics (service="kubelet" for example). From my understanding, `workload_name` contains the name of the managed-app in question so like this we get a separate request and error metric for each of them.

Reply:
The service here is referring to the SLO (Service Level Objective) and has nothing to do with the Kubernetes Service. We could theoretically have a service for efk-curator and efk-clusters for example

Reply:
For now, renaming workload name should be sufficient but you might want different SLOs per service inside an app for example

Reply:
We need to find something else for such cases. The service name or component could be coming from an annotation for example